cmd/ceremony/main.go

```diff
@@ -258,6 +259,75 @@ func (orc ocspRespConfig) validate() error {
 	return nil
 }
 
+type crlConfig struct {
+	CeremonyType string `yaml:"ceremony-type"`
+	PKCS11       struct {
+		Module       string `yaml:"module"`
+		PIN          string `yaml:"pin"`
+		SigningSlot  uint   `yaml:"signing-key-slot"`
+		SigningLabel string `yaml:"signing-key-label"`
+		SigningKeyID string `yaml:"signing-key-id"`
+	} `yaml:"pkcs11"`
+	Inputs struct {
+		IssuerCertificatePath string `yaml:"issuer-certificate-path"`
+	} `yaml:"inputs"`
+	Outputs struct {
+		CRLPath string `yaml:"crl-path"`
+	} `yaml:"outputs"`
+	CRLProfile struct {
+		ThisUpdate          string `yaml:"this-update"`
+		NextUpdate          string `yaml:"next-update"`
+		Number              int64  `yaml:"number"`
+		RevokedCertificates []struct {
+			CertificatePath string `yaml:"certificate-path"`
+			RevocationDate  string `yaml:"revocation-date"`
+		} `yaml:"revoked-certificates"`
+	} `yaml:"crl-profile"`
+}
+
+func (cc crlConfig) validate() error {
+	// PKCS11 fields
+	if cc.PKCS11.Module == "" {
+		return errors.New("pkcs11.module is required")
+	}
+	// key-slot cannot be tested because 0 is a valid slot
+	if cc.PKCS11.SigningLabel == "" {
+		return errors.New("pkcs11.signing-key-label is required")
+	}
+	if cc.PKCS11.SigningKeyID == "" {
+		return errors.New("pkcs11.signing-key-id is required")
+	}
+
+	// Input fields
+	if cc.Inputs.IssuerCertificatePath == "" {
+		return errors.New("inputs.issuer-certificate-path is required")
+	}
+
+	// Output fields
+	if cc.Outputs.CRLPath == "" {
+		return errors.New("outputs.crl-path is required")
+	}
+
+	// CRL profile fields
+	if cc.CRLProfile.ThisUpdate == "" {
+		return errors.New("crl-profile.this-update is required")
+	}
+	if cc.CRLProfile.NextUpdate == "" {
+		return errors.New("crl-profile.next-update is required")
+	}
+	// CRL Number cannot be tested because 0 is a valid CRL number
```

Comment:
I think it would be better to just make it a requirement that CRL sequences generated with this tool always start with 1.

Reply:
I think this still needs to be implemented - checking that CRLNumber != 0 (and updating the comment).